rickroll.py

```diff
@@ -153,5 +174,55 @@ def original():
 
     return str(play_tune(tune))
 
+@app.route("/sms", methods=["POST"])
+def sms():
+    """Adds an incoming SMS to a queue, to reply to later."""
+    bucket.put_object(
+        Key='queue/{to}/{from_}'.format(to=request.form['To'],
+                                        from_=request.form['From']),
+        Body='',
+    )
+    return "Hello world!"
+
+def send_sms():
+    """Empties the queue of incoming SMSes, replying to each one."""
+    for queue_entry in bucket.objects.filter(Prefix='queue/'):
+        _, our_number, their_number = queue_entry.key.split("/")
+        state_key = "state/{}/{}".format(our_number, their_number)
+
+        # Find out which message we sent last, if any
+        # Error handling taken from https://stackoverflow.com/a/33843019
+        try:
```

Comment:
Code climate is point out that this function [is a bit complex](https://codeclimate.com/github/pjf/rickastley/pull/7), and this seems perfect for factoring out to a separate function which returns the next message number (or raises on error).

I'm guessing that if we do throw an exception, we should log that but still try to process the other messages in our queue, so a single bad message doesn't knock out the entire SMS queue. (This isn't a blocker for this PR, but indicates future work that may be required.)

I have no idea what the best way is to log an error from Lambda.